### ISSUE TITLE
[Bug]: Cannot include `package.json` (needed for resolution) in `srcs` when `resolve_json_module = True`

### ISSUE THREAD

**Issue by Silic0nS0ldier** [aspect-build/rules_ts#689]
### What happened?

The target resembles

```starlark
ts_project(
    name = "ts_prod",
    srcs = glob([
        "src/**/*.ts",
    ]) + [
        # need the package.json so TS can resolve the type: "module"
        ":package.json",
    ],
    resolve_json_module = True,
    # ...
)
```

When the target is analysed the following error is raised

```
ERROR: Traceback (most recent call last):
        File "/___/___/BUILD.bazel", line 21, column 11, in <toplevel>
                ts_project(
        File "/___/bazel_output_root/a0311ccbc25922c56ee2769278435101/external/aspect_rules_ts~/ts/defs.bzl", line 378, column 20, in ts_project
                ts_project_rule(
Error in ts_project: rule 'ts_prod' has file 'package.json' as both an input and an output
ERROR: package contains errors: ___: Traceback (most recent call last):
        File "/___/___/BUILD.bazel", line 21, column 11, in <toplevel>
                ts_project(
        File "/___/bazel_output_root/a0311ccbc25922c56ee2769278435101/external/aspect_rules_ts~/ts/defs.bzl", line 378, column 20, in ts_project
                ts_project_rule(
Error in ts_project: rule 'ts_prod' has file 'package.json' as both an input and an output
```

This does not occur in Rules TS v2.2.0, and only occurs in conjunction with `resolve_json_module = True`.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: `7.3.0`

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file: `3.1.0`

Language(s) and/or frameworks involved: NA

### How to reproduce

```shell
1. Define a target with `ts_project` that takes `package.json` as an explicit input and `resolve_json_module = True`.
2. Perform any action which results in analysis of the containing `BUILD` file.
```

### Any other information?

`resolve_json_module = True` is _not_ used to import file listed in `srcs` with this scenario, rather this is being used to retrieve package version information in a strictly typed manner.

```ts
import eslintPackageJson from 'eslint/package.json' with { type: 'json' };
```

**Comment by jbedard** [comment]
My attempt to reproduce it at HEAD has failed. Can you checkout https://github.com/aspect-build/rules_ts/pull/711 and see if you can reproduce it at HEAD or confirm it's fixed?

**Comment by jbedard** [comment]
I think this only occurs when you references files via label syntax such as `:package.json`. Is this still an issue for you?
